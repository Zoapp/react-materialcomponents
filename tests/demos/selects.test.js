import React from "react";
import renderer from "react-test-renderer";
import Selects from "demos/components/selects";

describe("demos/components/selects", () => {
  it("renders correctly", () => {
    const component = renderer.create(<Selects />).toJSON();
    expect(component).toMatchSnapshot();
  });
});
