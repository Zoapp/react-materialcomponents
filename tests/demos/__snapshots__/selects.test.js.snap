// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`demos/components/selects renders correctly 1`] = `
<section>
  <h1>
    Select examples 
  </h1>
  <div
    style={
      Object {
        "padding": "16px",
      }
    }
  >
    <div
      aria-hidden={true}
      className="mdc-select"
      onClick={[Function]}
      onKeyUp={[Function]}
      role="listbox"
      style={Object {}}
    >
      <div
        className="mdc-select__surface"
        tabIndex="0"
      >
        <div
          className="mdc-select__label"
        >
          Pick a Food Group
        </div>
        <div
          className="mdc-select__selected-text"
        />
        <div
          className="mdc-select__bottom-line"
        />
      </div>
      <div
        className="mdc-select__menu mdc-menu"
        tabIndex="-1"
      >
        <ul
          aria-hidden="true"
          className="mdc-menu__items mdc-list"
          role="menu"
        >
          <li
            className="mdc-list-item"
            onClick={[Function]}
            onKeyUp={[Function]}
            role="option"
            tabIndex="0"
          >
            read, Cereal, Rice, and Pasta
          </li>
          <li
            className="mdc-list-item"
            onClick={[Function]}
            onKeyUp={[Function]}
            role="option"
            tabIndex="0"
          >
            Vegetables
          </li>
          <li
            className="mdc-list-item"
            onClick={[Function]}
            onKeyUp={[Function]}
            role="option"
            tabIndex="0"
          >
            Fruit
          </li>
          <li
            className="mdc-list-item"
            onClick={[Function]}
            onKeyUp={[Function]}
            role="option"
            tabIndex="0"
          >
            Milk, Yogurt, and Cheese
          </li>
          <li
            className="mdc-list-item"
            onClick={[Function]}
            onKeyUp={[Function]}
            role="option"
            tabIndex="0"
          >
            Meat, Poultry, Fish, Dry Beans, Eggs, and Nuts
          </li>
          <li
            className="mdc-list-item"
            onClick={[Function]}
            onKeyUp={[Function]}
            role="option"
            tabIndex="0"
          >
            Fats, Oils, and Sweets
          </li>
        </ul>
      </div>
    </div>
  </div>
  <h2>
    SelectField Properties
  </h2>
  <div
    className="mdc-layout-grid"
  >
    <div
      className="mdc-layout-grid__inner"
    >
      <div
        className="mdc-layout-grid__cell"
      >
        Name
      </div>
      <div
        className="mdc-layout-grid__cell"
      >
        Default
      </div>
      <div
        className="mdc-layout-grid__cell"
      >
        Type
      </div>
    </div>
    <div
      className="mdc-layout-grid__inner"
      style={
        Object {
          "borderBottom": "solid black 1px",
          "marginBottom": "16px",
          "paddingTop": "5px",
        }
      }
    >
      <div
        className="mdc-layout-grid__cell"
      >
        children
      </div>
      <div
        className="mdc-layout-grid__cell"
      >
        *
      </div>
      <div
        className="mdc-layout-grid__cell"
      >
        string
      </div>
    </div>
    <div
      className="mdc-layout-grid__inner"
      style={
        Object {
          "borderBottom": "solid black 1px",
          "marginBottom": "16px",
          "paddingTop": "5px",
        }
      }
    >
      <div
        className="mdc-layout-grid__cell"
      >
        mdcElement
      </div>
      <div
        className="mdc-layout-grid__cell"
      >
        MDC_SELECT
      </div>
      <div
        className="mdc-layout-grid__cell"
      >
        string
      </div>
    </div>
    <div
      className="mdc-layout-grid__inner"
      style={
        Object {
          "borderBottom": "solid black 1px",
          "marginBottom": "16px",
          "paddingTop": "5px",
        }
      }
    >
      <div
        className="mdc-layout-grid__cell"
      >
        disabled
      </div>
      <div
        className="mdc-layout-grid__cell"
      >
        false
      </div>
      <div
        className="mdc-layout-grid__cell"
      >
        bool
      </div>
    </div>
    <div
      className="mdc-layout-grid__inner"
      style={
        Object {
          "borderBottom": "solid black 1px",
          "marginBottom": "16px",
          "paddingTop": "5px",
        }
      }
    >
      <div
        className="mdc-layout-grid__cell"
      >
        label
      </div>
      <div
        className="mdc-layout-grid__cell"
      >
        null
      </div>
      <div
        className="mdc-layout-grid__cell"
      >
        string
      </div>
    </div>
    <div
      className="mdc-layout-grid__inner"
      style={
        Object {
          "borderBottom": "solid black 1px",
          "marginBottom": "16px",
          "paddingTop": "5px",
        }
      }
    >
      <div
        className="mdc-layout-grid__cell"
      >
        onSelected
      </div>
      <div
        className="mdc-layout-grid__cell"
      >
        null
      </div>
      <div
        className="mdc-layout-grid__cell"
      >
        function
      </div>
    </div>
    <div
      className="mdc-layout-grid__inner"
      style={
        Object {
          "borderBottom": "solid black 1px",
          "marginBottom": "16px",
          "paddingTop": "5px",
        }
      }
    >
      <div
        className="mdc-layout-grid__cell"
      >
        selectedIndex
      </div>
      <div
        className="mdc-layout-grid__cell"
      >
        -1
      </div>
      <div
        className="mdc-layout-grid__cell"
      >
        number
      </div>
    </div>
    <div
      className="mdc-layout-grid__inner"
      style={
        Object {
          "borderBottom": "solid black 1px",
          "marginBottom": "16px",
          "paddingTop": "5px",
        }
      }
    >
      <div
        className="mdc-layout-grid__cell"
      >
        style
      </div>
      <div
        className="mdc-layout-grid__cell"
      >
        {}
      </div>
      <div
        className="mdc-layout-grid__cell"
      >
        shape
      </div>
    </div>
  </div>
</section>
`;
